proposals/lip-0055.md

```diff
@@ -400,6 +400,24 @@ unsignedBlockHeaderSchema = {
 }
 ```
 
+
+#### Block Signature Calculation
+
+The block header signature is calculated using the `calculateBlockSignature` function. This function returns a bytes value with length `SIGNATURE_LENGTH_BYTES`.
+
+```python
+calculateBlockSignature():
+  generatorKey = validators.getValidatorAccount(block.header.generatorAddress).generatorKey
+  let secretKey be the private key corresponding to the public generatorKey
+
+  # Serialize the block header without signature
+  let serializedUnsignedBlockHeader be the serialization of block.header following the unsignedBlockHeaderSchema
+
+  let networkIdentifier be the network identifier of the chain
+
+  return signMessage(secretKey, "LSK_BH_", networkIdentifier, serializedUnsignedBlockHeader)
```

Comment:
I would leave a reference to LIP 37 regarding `signMessage` as it is done for `verifyMessageSig`

Reply:
added